src/lib/support/CHIPArgParser.cpp

```diff
@@ -81,9 +82,14 @@ static bool HelpTextContainsLongOption(const char * optName, const char * helpTe
 static bool HelpTextContainsShortOption(char optChar, const char * helpText);
 #endif // CHIP_CONFIG_ENABLE_ARG_PARSER_SANTIY_CHECK
 
+static inline bool IsASCII(int ch)
+{
+    return (ch >= 0 && ch < 128) ? true : false;
+}
+
 static inline bool IsShortOptionChar(int ch)
 {
-    return isascii(ch) && isgraph(ch);
+    return IsASCII(ch) && isgraph(ch);
```

Comment:
recommend you drop is isascii() check, it is superfluous.

Reply:
```
#include <ctype.h>
#include <stdio.h>

int main()
{
    int cols = 0;
    for (int i = 0; i < 0x7fffffff; i++)
    {
        if (isgraph(i))
        {
            printf("%c ", (char) i);
            cols += 1;
            cols &= 15;
            if (!cols)
            {
                printf("\n");
            }
        }
    }
    printf("\ndone\n");
}
```
yeilds:
```
g++ -o isgraph isgraph.cpp && ./isgraph
! " # $ % & ' ( ) * + , - . / 0 
1 2 3 4 5 6 7 8 9 : ; < = > ? @ 
A B C D E F G H I J K L M N O P 
Q R S T U V W X Y Z [ \ ] ^ _ ` 
a b c d e f g h i j k l m n o p 
q r s t u v w x y z { | } ~ 
done
```

Reply:
Right. Even after removing isascii, I decided to leave IsShortOptionChar as I think that using it in checks is more meaningful than calling isgraph directly and compiler will optimize it anyway. Do you agree?